packages/server/src/migrations/0.1.18-status.ts

```diff
@@ -0,0 +1,35 @@
+import { Migration } from '@botpress/messaging-engine'
+
+export class StatusMigration extends Migration {
+  meta = {
+    name: StatusMigration.name,
+    description: 'Moves status information from conduit table to the status table',
+    version: '0.1.19'
+  }
+
+  async valid() {
+    return this.trx.schema.hasTable('msg_conduits')
+  }
+
+  async applied() {
+    return !(await this.trx.schema.hasColumn('msg_conduits', 'initialized'))
+  }
+
+  async up() {
+    await this.trx.schema.dropTableIfExists('msg_status')
+
+    return this.trx.schema.alterTable('msg_conduits', (table) => {
+      table.dropIndex(['initialized'])
+      table.dropColumn('initialized')
+    })
+  }
+
+  async down() {
+    await this.trx.schema.dropTableIfExists('msg_status')
+
+    return this.trx.schema.alterTable('msg_conduits', (table) => {
+      table.timestamp('initialized').nullable()
+      table.index(['initialized'])
+    })
```

Comment:
I understand that `applied` protects the up migration in case it was already executed but there is no similar mechanism for the down migration. 

I think we should maybe pass an argument to applied (up/down) and execute different queries depending on the context. What do you think? The same could apply to valid!

Reply:
applied is use for both up and down. This is the code for the base migration class. We just inverse the condition depending if it's up or down :

```ts
  async shouldRun() {
    if (!(await this.valid())) {
      return false
    }

    if (this.isDown) {
      return this.applied()
    } else {
      return !(await this.applied())
    }
  }
```